### PR TITLE
Rules --> Add top-level Examples gallery

### PR DESCRIPTION
**Pull request by kodster28**
### Summary

- [x] [Stream] Massage and metadata
- [x] Other products --> Pull into alignment to keep current behavior
- [x] Rules --> Pull everything together at the top level

### How to review

- Check out the main [Rules examples](https://examples-tweaks.cloudflare-docs-7ou.pages.dev/rules/examples/) page.
- Check out sub-pages w/in the Rules to make sure only those [examples are showing](https://examples-tweaks.cloudflare-docs-7ou.pages.dev/rules/snippets/examples/)
- Review individual examples page to check that the behavior is better or the same as previous (Stream / Vectorize / Queues / D1 / etc).

### CODE REVIEW

File: src/content/docs/r2/examples/aws/aws-cli.mdx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: aws CLI
-pcx_content_type: configuration
+pcx_content_type: example
```
Comment (marciocloudflare) [inline src/content/docs/r2/examples/aws/aws-cli.mdx:3]: The overview for R2 examples doesn't render like the other ones above (with that new card component). Is this on purpose?

File: src/content/docs/rules/snippets/examples/ab-testing-same-url.mdx [hunk 2]
```diff
@@ -20,37 +19,37 @@ This version passes through requests for `/test/*` and `/control/*` URI paths to
 const NAME = "myExampleABTest";
 
 export default {
-  async fetch(request) {
-    // Clone the original URL
-    const url = new URL(request.url);
+	async fetch(request) {
```
Comment (marciocloudflare) [inline src/content/docs/rules/snippets/examples/ab-testing-same-url.mdx:22]: Does this need so much spacing?

File: src/content/docs/rules/transform/examples/add-request-header-bot-score.mdx [hunk 3]
```diff
@@ -2,17 +2,16 @@
 pcx_content_type: example
 summary: Create a request header modification rule (part of Transform Rules) to
   add a `X-Bot-Score` HTTP header to the request with the current bot score.
-product:
+products:
```
Comment (marciocloudflare) [inline src/content/docs/rules/transform/examples/add-request-header-bot-score.mdx:5]: The Example menu in Transform Rules isn't a dropdown, so when you open an example there's no way to tell where it is, and no way to quickly see all examples in the sidenav (like in all other Examples).  Is this a mistake?

**Review by marciocloudflare — APPROVED** [review]

File: src/content/docs/rules/snippets/examples/ab-testing-same-url.mdx (hunk 2 above)
Comment (kodster28) [reply src/content/docs/rules/snippets/examples/ab-testing-same-url.mdx:22]: I think it's the new default `tab` formatting introduced as part of https://github.com/cloudflare/cloudflare-docs/blob/production/.editorconfig

File: src/content/docs/r2/examples/aws/aws-cli.mdx (hunk 1 above)
Comment (kodster28) [reply src/content/docs/r2/examples/aws/aws-cli.mdx:3]: Yeah, it's matching how it looks currently (though good catch!).

**Review by KianNH — APPROVED** [review]

**Pull request merged**